docs/designs/rubygems-001.md

```diff
@@ -0,0 +1,547 @@
+exists, it needs to be explicitly imported in the `Gemfile`:
+
+```
+source "https://rubygems.org"
+
+gemspec
+```
+
+When the `.gemspec` file exists and is properly imported, it will be listed in the `Gemfile.lock` as a `PATH`
+dependency:
+
+```
+PATH
+  remote: .
+  specs:
+    tmp (0.1.2)
+```
+
+Since the `remote` will always point to `.` in case of the main package [^main-package], we can safely use it to get
+the `name` and `version` for the SBOM component. In case this block is absent, we will need to fallback to the
+repository's remote `origin` to retrieve the main package's name, and leave the version empty, since it is not a
+mandatory field.
+
+[^main-package]: In Cachi2's terms, the **main package** is the path in the repository that is currently being
+  processed.
+
+##### PURLs
+
+Also check the Ruby PURL [specification](https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#gem).
+
+###### Standard Gem
+```txt
+pkg:gem/my-gem-name@0.1.1
+```
+
+###### Git dependency
+
+```txt
+pkg:gem/my-git-dependency?vcs_url=git%2Bhttps://github.com/my-org/mygem.git%26487618a68443e94d623bb585cb464b07d36702
+```
+
+The metadata for a Git dependency can be read from the `Gemfile.lock`:
+
+```
+GIT
+  remote: https://github.com/my-org/mygem.git
+  revision: 26487618a68443e94d623bb585cb464b07d36702
+  specs:
+    json-schema (3.0.0)
+      addressable (>= 2.4)
+```
+
+###### Path dependency
+
+```txt
+pkg:gem/my-path-dependency?vcs_url=git%2Bhttps://github.com/my-org/mygem.git%40b6f47bd07e669c8d2eced8015c4bfb06db49949#subpath
+```
+
+The PURL is formed by the current repository remote origin URL and ref, and the subpath that is specified in the
+`Gemfile.lock`:
+
+```ruby
+PATH
+  remote: subpath
+  specs:
+    my-path-dependency (1.0.0)
+```
+
+## Decision
+
+The support for Rubygems will be implemented as currently described in this design document.
+
+### Scoping of the initial implementation
+- design high-level code structure into multiple modules
+- create a test repository that contains all the relevant use cases
+- define models for RubyGems as the new package manager
+- parse all gems from `Gemfile.lock`
+- implement metadata parsing for the "main package"
+- download all gems from rubygems.org
+- download all gems from Git repositories
+- validate path dependencies are relative to the project root
+- inject the Bundler configuration needed for the offline install
+- generate PURLs for all dependencies
+- add integration and e2e tests
+- add documentation
+
+### Potential follow-up features
+- implement checksum parsing and validation when prefetching from the registry
+- downloading the Bundler version specified in the `Gemfile.lock`
+- support for pre-compiled binaries (platforms other than `ruby`)
+- Gemfile.lock checksum validation (blocked by pending official support)
+- reporting dev dependencies
+- proper support for plugins
```

Comment:
This is just a summary of the implementation specification and as such is IMO redundant as it provides very little information compared to the specification section.

Reply:
I think the summary is really... just a summary. It's a way to help the reader bringing to mind all the info that was more thoroughly explained before. It can be optional as well, as I don't see it being useful for smaller docs.

There is also another benefit for this specific section here: clear separation of what's in scope and what's not. But maybe scoping doesn't belong to the design doc.

Reply:
> There is also another benefit for this specific section here: clear separation of what's in scope and what's not. But maybe scoping doesn't belong to the design doc.

arguably it does, because a given pkg manager is a massive ecosystem with loads of variables and corner cases, so the design doc needs to outline what is being done and what's not, it just needs an appropriate format. I'd refrain from mentioning follow-ups as those IMO don't belong to a formal docs, one's imagination can be very wild when it comes to follow-ups so let the future decide what the follows are going to be. For the rest, an optional summary section would IMO be acceptable, worst case scenario, we'll decide it was wrong and drop it and propose a refined version of the template....RFC did the same...not once

Reply:
The follow-ups mentioned here are either things that were mentioned at some point in the design, but were marked as out-of-scope since we want to keep the implementation very basic. They are all features that have been similarly implemented in other package managers.

This is part of the summary, as I see it: "the design touches all these features, this is the subset we want to implement, and this is the subset we don't want to for the first version". Maybe we can just call the section as "out of the initial scope".

Reply:
I feel like there's a benefit to it... it clearly states, in a summarized form, what the initial deliverable is.